### PR TITLE
feat: subscribe to 2 long lived subnets per node

### LINKED ISSUE #5546: Subnet backbone structure based on beacon nodes
### Problem description

Right now each node has to subscribe to `RANDOM_SUBNETS_PER_VALIDATOR * validators` random (long lived) subnets. This results in significant bandwidth use for some nodes while other nodes may not follow this rule

### Solution description

Enforce all nodes to subscribe to `SUBNETS_PER_NODE` ~~random~~ deterministic subnets (2 for mainnet), this would distribute load to all beacon nodes

### Additional context

See https://github.com/ethereum/consensus-specs/pull/3312

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

- See https://github.com/ethereum/consensus-specs/pull/3312
- For Lodestar users, this helps reduce the network bandwidth, blocks will come to node more timely and there would be less missed attestations

**Description**

- Currently we subscribe to n long lived & random subnets per n validator
- This changes to consistently subscribe to `SUBNETS_PER_NODE` (2) long lived subnet based on node id
- Implement another `IAttnetsService` called `DLLAttnetsService` for this
- Add `--deterministicLongLivedAttnets` feature flag for this, disabled by default

Closes #5546

### CODE REVIEW

**Comment by twoeths** [comment]
since there are only 2 long lived subnets anymore, node has to unsubscribe and resubscribe a lot which make it unstable

need to wait for #5506 and a couple of new gossipsub PRs regarding `backoff` time

**Comment by twoeths** [comment]
with new libp2p, this branch has stable peers, but it's not as stable as "unstable" branch because node has to subscribe/unsubscribe a lot based on duties.

- lg1k has same performance because there are 1k validators
Screenshot 2023-07-11 at 08 53 52

Screenshot 2023-07-11 at 08 54 06

Screenshot 2023-07-11 at 08 56 19


- md16 has better missed ratio because it connects to less subnet topics

Screenshot 2023-07-11 at 08 54 41

Screenshot 2023-07-11 at 08 54 50

Screenshot 2023-07-11 at 08 56 39

**Comment by twoeths** [comment]
this is ready for review

File: packages/beacon-node/src/network/core/networkCore.ts [hunk 1]
```diff
@@ -185,7 +188,18 @@ export class NetworkCore implements INetworkCore {
...
+
+    await reqResp.start();
+    // should be called before DLLAttnetsService constructor so that node subscribe to deterministic attnet topics
+    await gossip.start();
+
+    const enr = opts.discv5?.enr;
+    const nodeId = enr ? fromHexString(ENR.decodeTxt(enr).nodeId) : null;
+    const attnetsService = opts.deterministicLongLivedAttnets
+      ? new DLLAttnetsService(config, clock, gossip, metadata, logger, metrics, nodeId, opts)
+      : new AttnetsService(config, clock, gossip, metadata, logger, metrics, opts);
```
Comment (wemeetagain) [inline packages/beacon-node/src/network/core/networkCore.ts:202]: :+1:

**Review by wemeetagain — APPROVED** [review]
LGTM, as you say, this is disabled by default. We can do more testing before turning this on.

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.10.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.10.0) :tada:

**Pull request merged**